tb_pulumi/s3.py

```diff
@@ -93,6 +111,162 @@ def __init__(
+          <https://www.pulumi.com/registry/packages/aws/api-docs/s3/bucketwebsiteconfigurationv2/>`_ defining the
+          operating parameters of the website.
+
+    :param name: A string identifying this set of resources.
+    :type name: str
+
+    :param project: The ThunderbirdPulumiProject to add these resources to.
+    :type project: tb_pulumi.ThunderbirdPulumiProject
+
+    :param bucket_name: The name of the S3 bucket to host a public website in.
+    :type bucket_name: str
+
+    :param website_config: A dict of options describing a `BucketWebsiteConfigurationV2 resource
+        <https://www.pulumi.com/registry/packages/aws/api-docs/s3/bucketwebsiteconfigurationv2/#inputs>`_ .
+    :type website_config: dict
+
+    :param opts: Additional pulumi.ResourceOptions to apply to these resources. Defaults to None.
+    :type opts: pulumi.ResourceOptions, optional
+
+    :param tags: Key/value pairs to merge with the default tags which get applied to all resources in this group.
+        Defaults to {}.
+    :type tags: dict, optional
+
+    :param kwargs: Additional arguments to pass into the :py:class:`S3Bucket` constructor.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        project: tb_pulumi.ThunderbirdPulumiProject,
+        bucket_name: str,
+        website_config: dict,
+        opts: pulumi.ResourceOptions = None,
+        tags: dict = {},
+        **kwargs,
+    ):
+        super().__init__('tb:s3:S3BucketWebsite', name=name, project=project, opts=opts, tags=tags)
+
+        bucket = S3Bucket(
+            f'{name}-bucket',
+            project=project,
+            bucket_name=bucket_name,
+            enable_server_side_encryption=False,
+            enable_versioning=False,
+            exclude_from_project=True,
+            tags=self.tags,
+            opts=pulumi.ResourceOptions(parent=self),
+            **kwargs,
+        )
+
+        # Required or else we can't place an ACL on the bucket
+        bucket_oc = aws.s3.BucketOwnershipControls(
+            f'{name}-bucket-oc',
+            bucket=bucket_name,
+            rule={'objectOwnership': 'ObjectWriter'},
+            opts=pulumi.ResourceOptions(parent=self, depends_on=[bucket]),
+        )
+
+        # Required or else we can't apply ACLs or a bucket policy, which are required elements of an S3 website
+        bucket_pab = aws.s3.BucketPublicAccessBlock(
+            f'{name}-bucket-pab',
+            bucket=bucket_name,
+            block_public_acls=False,
+            block_public_policy=False,
+            opts=pulumi.ResourceOptions(parent=self, depends_on=[bucket]),
+        )
+
+        bucket_acl = aws.s3.BucketAclV2(
+            f'{name}-bucket-acl',
+            bucket=bucket_name,
+            acl='public-read',
+            opts=pulumi.ResourceOptions(parent=self, depends_on=[bucket, bucket_oc, bucket_pab]),
+        )
+
+        policy_json = tb_pulumi.constants.IAM_POLICY_DOCUMENT.copy()
```

Comment:
Almost wonder if it's worth storing IAM_POLICY_DOCUMENT in a lambda function that returns a dict. Therefore we don't have to remember to run .copy().